src/platform/device-manager.ts

```diff
@@ -3,23 +3,75 @@
  * TODO(cjdaly) integrate with Mock Local Home Platform and Mock Network
  **/
 /// <reference types="@google/local-home-sdk" />
+export class DeviceManagerStub implements smarthome.DeviceManager {
+  /** Action to call when an `IntentRequest` is marked with `markPending()` */
+  private markPendingAction:
+    | ((request: smarthome.IntentRequest) => void)
+    | undefined;
+  /** Map of each expected `CommandRequest` to its associated response*/
+  private expectedCommandToResponse: Map<
+    smarthome.DataFlow.CommandRequest,
+    smarthome.DataFlow.CommandBase
+  > = new Map();
 
-import { DeviceManager } from '@google/local-home-sdk';
+  /**
+   * Checks if the next request marked pending matches a given request.
+   * @param requestToMatch  A request to test against the next request marked pending.
+   * @returns  Promise that resolves to a boolean representing if the requests matched.
```

Comment:
It will hang if `markPending()` is never called, but will resolve with `false` if there was a call without a match.  `ava` detects hanging promises with a `Promise returned by test never resolved` message.  It would be better if I could explicitly reject it, but I'm not sure at what point I should deem the `Promise` rejected / timed out.

Reply:
If `pendingRequest === requestToMatch` is `false`, should the promise reject?

Reply:
Done